service/service.py

```diff
@@ -279,6 +279,33 @@ def cancel_item(order_id, item_id):
     return make_response(jsonify(order.serialize()), status.HTTP_200_OK)
 
 
+######################################################################
+# SHIP AN ITEM IN AN ORDER
+######################################################################
+@app.route("/orders/<int:order_id>/items/<int:item_id>/ship", methods=["PUT"])
+def ship_item(order_id, item_id):
+    """ Ship a single item in the Order that have not being cancelled or delivered yet """
+    app.logger.info("Request to ship item with id: %s in order with id: %s", item_id, order_id)
+    order = Order.find(order_id)
+    if not order:
+        raise NotFound("Order with id '{}' was not found.".format(order_id))
+
+    order_item_found = False
```

Comment:
We should try to maintain a single-level of abstraction in our code - It would be prudent to write a separate function "find_order_item" which takes in a list of order items and an order item id to search, and returns the order item with the given order item id.

Here's a useful link to understand the single level of abstraction principle: https://8thlight.com/blog/javier-garc%C3%ADa/2019/06/11/refactoring-levels-of-abstraction.html

Reply:
I got what you mean. Thanks for the suggestions and the reference you shared is very helpful for my understanding! I'll modify my code now, and let you know once I finish!

Reply:
Hi Rutivk, I just updated my code! If you have a chance, could you please review my changes?

Reply:
Thanks. Done!